src/main/java/core/basesyntax/dao/animal/AnimalDaoImpl.java

```diff
@@ -12,11 +19,38 @@ public AnimalDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Animal save(Animal animal) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(animal);
+            transaction.commit();
+            return animal;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert animal Entity with name = "
+                    + animal.getName(), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Animal> findByNameFirstLetter(Character character) {
-        return null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Animal> criteriaQuery = criteriaBuilder.createQuery(Animal.class);
+            Root<Animal> root = criteriaQuery.from(Animal.class);
+            String str = character + "%";
+            criteriaQuery.select(root).where(criteriaBuilder.like(root.get("name"), str));
```

Comment:
criteriaQuery.select(root).where(criteriaBuilder.like(root.get("name"), character + "%"));
I think that is better